lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -32,384 +57,165 @@ def get(section: str, option: str, fallback=None) -> str:
+        assert strict
+        for t in datasets:
+            if self.path == t.tarball:
+                if t.dataset.resource_id in __class__.fails:
+                    break  # Drop out to raise an exception
+                return MockPath(self.path)
+        raise FileNotFoundError(f"No such file or directory: '{self.path}'")
+
+    def stat(self) -> StatResult:
+        for t in datasets:
+            if self.path == t.tarball:
+                return StatResult(st_mode=0o554, st_size=t.size)
+        raise FileNotFoundError(f"No such file or directory: '{self.path}'")
+
+    @classmethod
+    def _fail_on(cls, fails: list[str]):
+        cls.fails.extend(fails)
+
+    @classmethod
+    def _reset(cls):
+        cls.fails.clear()
+
+
+class MockSync:
+
+    record: dict[str, JSONOBJECT] = {}
+
+    def __init__(self, logger: Logger, component: str):
+        self.component = component
+        self.logger = logger
+
+    def next(self, operation: Operation) -> list[Dataset]:
+        return [x.dataset for x in datasets]
+
+    def update(self, dataset: Dataset, did: Operation, enabled: list[Operation]):
+        __class__.record[dataset.resource_id] = {"did": did, "enabled": enabled}
+
+    @classmethod
+    def _reset(cls):
+        cls.record = {}
+
+
+class MockCacheManager:
+
+    fails: list[str] = []
+    unpacked: list[str] = []
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        self.config = config
+        self.logger = logger
+
+    def unpack(self, id: str):
+        if id in __class__.fails:
+            for d in datasets:
+                if d.dataset.resource_id == id:
+                    raise TarballUnpackError(Path(d.tarball), "test error")
+            else:
+                raise Exception("I'm lost; where are you?")
+        __class__.unpacked.append(id)
+
+    @classmethod
+    def _fail_on(cls, fails: list[str]):
+        cls.fails.extend(fails)
+
+    @classmethod
+    def _reset(cls):
+        cls.fails.clear()
+        cls.unpacked.clear()
+
+
 class TestUnpackTarballs:
-    tarlist = ["/mock/ABC/TO-UNPACK/A.tar.xz", "/mock/ABC/TO-UNPACK/B.tar.xz"]
-    tar = Path(tarlist[0])
-    min_size, max_size = 0, 1000
-    tar_size, tar_mode = 586, 16893
-
-    def mock_stat(self, *args, follow_symlinks=False):
-        """Mocked replacement for Path.stat()"""
-        StatResult = namedtuple("StatResult", ["st_mode", "st_size"])
-        return StatResult(
-            st_mode=TestUnpackTarballs.tar_mode, st_size=TestUnpackTarballs.tar_size
-        )
+    @pytest.fixture()
+    def reset(self):
+        MockPath._reset()
+        MockSync._reset()
+        MockCacheManager._reset()
 
     @pytest.mark.parametrize(
-        "min_size, max_size, total", [(586, 1000, 2), (0, 1000, 2), (0, 586, 0)]
+        "min_size,max_size,targets",
+        [
+            (586, 1000, ["md5.3"]),
+            (0, 1000, ["md5.1", "md5.2", "md5.3"]),
+            (0, 586, ["md5.1", "md5.2"]),
+        ],
```

Comment:
The `586`'s should be `500`:  we're supposed to be testing when a dataset size hits a bucket boundary exactly.